niworkflows/interfaces/cifti.py

```diff
@@ -75,98 +82,135 @@ class GenerateCifti(SimpleInterface):
             self.inputs.bold_file,
             label_file,
             annotation_files,
-            self.inputs.gifti_files,
-            self.inputs.volume_target,
-            self.inputs.surface_target,
+            self.inputs.surface_files,
+            targets,
             self.inputs.TR)
         return runtime
 
     def _define_variant(self):
         """Assign arbitrary label to combination of CIFTI spaces"""
         space = None
         variants = {
-            # to be expanded once addtional spaces are supported
+            'fsLR': ['fsLR', 'MNI152NLin2009cAsym'],
             'space1': ['fsaverage5', 'MNI152NLin2009cAsym'],
             'space2': ['fsaverage6', 'MNI152NLin2009cAsym'],
         }
         for sp, targets in variants.items():
-            if all(target in targets for target in
-                    [self.inputs.surface_target, self.inputs.volume_target]):
+            if all(
+                target in targets for target in
+                [self.inputs.surface_target, self.inputs.volume_target]
+            ):
                 space = sp
         if space is None:
             raise NotImplementedError
 
         variant_key = os.path.abspath('dtseries_variant.json')
         with open(variant_key, 'w') as fp:
             json.dump({space: variants[space]}, fp)
-        return variant_key, space
+        self._results['variant_key'] = variant_key
+        self._results['variant'] = space
+        return variants[space]
 
     def _fetch_data(self):
         """Converts inputspec to files"""
-        if (self.inputs.surface_target == "fsnative" or
-                self.inputs.volume_target != "MNI152NLin2009cAsym"):
-            # subject space is not support yet
-            raise NotImplementedError
-
-        annotation_files = sorted(glob(os.path.join(self.inputs.subjects_dir,
-                                                    self.inputs.surface_target,
-                                                    'label',
-                                                    '*h.aparc.annot')))
-        if not annotation_files:
-            raise IOError("Freesurfer annotations for %s not found in %s" % (
-                          self.inputs.surface_target, self.inputs.subjects_dir))
-
-        label_file = str(get_template(
-            'MNI152NLin2009cAsym', resolution=2, desc='DKT31', suffix='dseg'))
+        if (
+            self.inputs.surface_target not in  ("fsaverage5", "fsaverage6", "fsLR") or
+            self.inputs.volume_target not in ("MNI152NLin2009cAsym", "MNI152NLin6Asym")
+        ):
+            raise NotImplementedError(
+                "Target space (surface: {0}, volume: {1}) is not supported".format(
+                    self.inputs.surface_target, self.inputs.volume_target
+                )
+            )
+
+        tpl_kwargs = {}
+        annotation_files = None
+        if self.inputs.surface_target.startswith("fsav"):
```

Comment:
I guess this decision should be made based on whether MNI6 OR MNI2009c was used?